// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

syntax = "proto2";
package hit.protobuf;

message Ciphertext {
	required int32 version = 1;
	required bool initialized = 2; // has this ciphertext been initialized?
	optional bytes seal_ct = 3;    // the underlying SEAL ciphertext
	required int32 he_level = 4;   // level of this ciphertext
	repeated double raw_pt = 5;    // raw (unscaled) plaintext encrypted in the ciphertext; only non-empty in debug modes
	required double scale = 6;     // CKKS scale of this ciphertext
}
